lib/network_monitor.c

```diff
@@ -758,6 +815,173 @@ static void insert_addr(struct mptcpd_nm *nm,
         (void) insert_addr_return(interface, rtm_addr);
 }
 
+static size_t rta_add_u32(void *rta_buf, unsigned short type, uint32_t value)
+{
+        struct rtattr *rta = rta_buf;
+
+        rta->rta_len = RTA_LENGTH(sizeof(uint32_t));
+        rta->rta_type = type;
+        *((uint32_t *) RTA_DATA(rta)) = value;
+
+        return RTA_SPACE(sizeof(uint32_t));
+}
```

Comment:
It comes from libell, but there aren't many different ways to implement this chunk of code. (e.g. iproute2 bits are quite similar). Did libell take this from iproute? 

AFAICS reading https://en.wikipedia.org/wiki/GNU_General_Public_License#Compatibility_and_multi-licensing mixing BSD-3 clause and LGPL2.1 is ok.

Otherwise I can rewrite the code...

Reply:
pushed a new version with the relevant helper re-written

Reply:
I don't know the origin of ELL's code, but that is moot with your rewrite.  Thanks for doing that!